### PR TITLE
Add benchmarks for native histograms

### PR DESCRIPTION
**Pull request by fpetkovski**
This commit adds benchmarks for queries on native histograms. The benchmark results should be added to the same output as the ones for queries on floats.

### CODE REVIEW

File: execution/aggregate/vector_table.go [hunk 1]
```diff
@@ -129,10 +129,10 @@ func newVectorAccumulator(expr parser.ItemType) (vectorAccumulator, error) {
 
 func histogramSum(histograms []*histogram.FloatHistogram) *histogram.FloatHistogram {
 	if len(histograms) == 1 {
-		return histograms[0]
+		return histograms[0].Copy()
```
Comment (saswatamcode) [inline execution/aggregate/vector_table.go:132]: Is there a reason for this change? Did this fix the test somehow? 🙂

Comment (fpetkovski) [reply execution/aggregate/vector_table.go:132]: Good question, it did in fact fix the test after I changed the number of generated histograms.

When applying a lookback, the selector can return the same instance of histogram as in previous steps. So if we don't copy the histogram for the sum result, we will end up modifying it in place when adding new histograms to it. Then when we apply a lookback, the returned histogram will actually be the `sum` result.

This is where the lookback is done: https://github.com/thanos-community/promql-engine/blob/main/execution/scan/vector_selector.go#L189.

Comment (saswatamcode) [reply execution/aggregate/vector_table.go:132]: Thanks! That makes sense

File: engine/engine_test.go [hunk 2]
```diff
@@ -3399,7 +3402,12 @@ func testNativeHistograms(t *testing.T, cases []histogramTestCase, opts promql.E
 
 						sortByLabels(promResult)
 						sortByLabels(newResult)
-						testutil.Equals(t, promVector, newVector)
+						if hasNaNs(promResult) {
+							t.Log("Applying comparison with NaN equality.")
+							testutil.WithGoCmp(cmpopts.EquateNaNs()).Equals(t, promVector, newVector)
+						} else {
+							require.Equal(t, promVector, newVector)
```
Comment (saswatamcode) [inline engine/engine_test.go:3409]: Let's maybe keep single testing lib
```suggestion
							testutil.Equals(t, promVector, newVector)
```

File: engine/engine_test.go [hunk 3]
```diff
@@ -3423,18 +3431,25 @@ func testNativeHistograms(t *testing.T, cases []histogramTestCase, opts promql.E
 							testutil.Assert(t, len(expected) == 0)
 						}
 						testutil.Equals(t, len(expected), len(actual))
-						testutil.Equals(t, expected, actual)
+						if hasNaNs(res) {
+							t.Log("Applying comparison with NaN equality.")
+							testutil.WithGoCmp(cmpopts.EquateNaNs()).Equals(t, expected, actual)
+						} else {
+							require.Equal(t, expected, actual)
```
Comment (saswatamcode) [inline engine/engine_test.go:3438]:
```suggestion
							testutil.Equals(t, expected, actual)
```

**Review by saswatamcode — APPROVED** [review]
LGTM! Thanks for adding this! Maybe we can add the results to README too?

Small suggestion.

**Comment by fpetkovski** [comment]
Good point, updated 👍

**Review by saswatamcode — APPROVED** [review]
LGTM

**Pull request merged**